derive_client/_bridge/transaction.py

```diff
@@ -51,6 +51,8 @@ def _increase_token_allowance(
 ) -> None:
     func = erc20_contract.functions.approve(spender, amount)
     tx = build_standard_transaction(func=func, account=from_account, w3=w3)
-    tx_result = send_and_confirm_tx(w3=w3, tx=tx, private_key=private_key, action="approve()", logger=logger)
-    if tx_result.status != TxStatus.SUCCESS:
+    signed_tx = sign_tx(w3=w3, tx=tx, private_key=private_key)
+    tx_hash = send_tx(w3=w3, signed_tx=signed_tx)
+    tx_receipt = wait_for_tx_finality(w3=w3, tx_hash=tx_hash, logger=logger)
+    if tx_receipt.status != TxStatus.SUCCESS:
         raise RuntimeError("approve() failed")
```

Comment:
This is still not great. It will execute this during preparation of the actual bridge transaction. because we want for finality, this will take a long time just to get the `PreparedBridgeTx` back.

We've discussed this and batching them for atomic execution will require a light account (or safe...)